pkg/secret/signing_secret.go

```diff
@@ -0,0 +1,96 @@
+package secret
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type KeyType int
+
+const (
+	CosignKeyType KeyType = iota
+	PKCS8KeyType
+)
+
+type SigningKey struct {
+	SecretName string
+	Key        []byte
+	Password   []byte
+	Type       KeyType
+}
+
+func FilterCosignSigningSecrets(secrets []*corev1.Secret) []*corev1.Secret {
+	return filterCosignSecrets(secrets, "")
+}
+
+func FilterAndExtractSLSASecrets(secrets []*corev1.Secret) ([]SigningKey, error) {
+	cosignSecrets := filterCosignSecrets(secrets, SLSASecretAnnotation)
+	privKeySecrets := filterPrivateKeySecrets(secrets, SLSASecretAnnotation)
+
+	return extractAttestationKeyFromSecrets(cosignSecrets, privKeySecrets)
+}
+
+func filterCosignSecrets(serviceAccountSecrets []*corev1.Secret, annotation string) []*corev1.Secret {
```

Comment:
Is this `filterCosignSecrets` or `gatherCosignSecrets`?  I'd expect a filter to apply a predicate to some collection.  Here it looks like we're gathering the secrets and packing them into a struct, should they exist.